### PR TITLE
[IOTDB-1990] Fix uncheck null result by calling IReaderByTimestamp

### PR DESCRIPTION
**Pull request by ericpai**
See JIRA: https://issues.apache.org/jira/browse/IOTDB-1990

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/query/dataset/RawQueryDataSetWithValueFilter.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ private boolean cacheRowRecords() throws IOException {
 
       // 3. use values in results to fill row record
       for (int j = 0; j < cachedTimeCnt; j++) {
-        if (results[j] == null) {
+        if (results == null || results[j] == null) {
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/query/dataset/RawQueryDataSetWithValueFilter.java:121]: It seems that only when `cachedTimeCnt` equals to zero, `getValuesInTimestamps` will return null. However, if `cachedTimeCnt` equals to zero, we will `return false` before here.

File: server/src/main/java/org/apache/iotdb/db/query/dataset/RawQueryDataSetWithValueFilter.java [hunk 2]
```diff
@@ -199,7 +199,7 @@ private boolean cacheRowInObjects() throws IOException {
 
       // 3. use values in results to fill row record
       for (int j = 0; j < cachedTimeCnt; j++) {
-        if (results[j] != null) {
+        if (results != null && results[j] != null) {
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/query/dataset/RawQueryDataSetWithValueFilter.java:202]: Same as above.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]
Could you provide an IT for this kind of case

**Comment by ericpai** [comment]
@JackieTien97 I replayed this test case `IoTDBUDTFAlignByTimeQueryIT.queryWithValueFilter9` in a cluster with 3 nodes and 2 replicas. It can reappear in 100%. This is the debug screenshot. I guess the reason is that although the `cachedTimeCnt` is not empty, there's no data matched the `cachedTimeCnt` and a null result returns. This only happens in cluster query.

image

**Pull request merged**